swarm.go

```diff
@@ -208,9 +208,6 @@ func (s *Swarm) addConn(tc transport.Conn, dir inet.Direction) (*Conn, error) {
 	// * The other will be decremented when Conn.start exits.
 	s.refs.Add(2)
 
-	// Take the notification lock before releasing the conns lock to block
-	// Disconnect notifications until after the Connect notifications done.
-	c.notifyLk.Lock()
```

Comment:
As noted in the comment, this will allow disconnect events to fire (concurrently) before connect events. This is an important invariant to maintain and has caused us no end of grief in the past.

Reply:
Are there unit tests for this?

Reply:
Not that I know of.